src/cis/fam_cis_thallium_client.cpp

```diff
@@ -546,8 +546,6 @@ void *Fam_CIS_Thallium_Client::copy(
     cisRequest.set_src_used_memsrv_cnt(srcUsedMemsrvCnt);
     cisRequest.set_src_keys(srcKeys, (int)srcUsedMemsrvCnt);
     cisRequest.set_src_base_addr(srcBaseAddrList, (int)srcUsedMemsrvCnt);
-    Fam_CIS_Thallium_Response cisResponse = rp_copy.on(ph)(cisRequest);
-    RPC_STATUS_CHECK(CIS_Exception, cisResponse)
 
     auto my_async_response_ptr =
         new tl::async_response(rp_copy.on(ph).async(cisRequest));
```

Comment:
Since we will no longer RPC status checks for copy, backup, restore, delete_backup and the tl::async_response procedure does not provide any methods to check if the response has successfully been queued, can we just check if the my_async_response_ptr is valid and if not, we can throw the CIS_Exception as in the wait calls here?

Reply:
That was fun! I got to learn so much about Thallium, Margo, and C++! (A bit sarcastic, but I did learn things. If I can just remember them.) So, if the async() part fails to hand off to Margo, then it will throw immediately; if not, the async_response will be created and if Margo has problems after that it will be detected and thrown when the wait is done.

Reply:
True that, what I see is that the async_response async() thallium procedure returns an iforward, which is  an async_response object that the caller can wait on. Here internally the MARGO_ASSERT() throws a margo_exception, which could be caught here with a try & catch (const tl::margo_exception& e) { err_msg }
This would let us handle the error before calling wait(). Please share you thoughts.

Reply:
It is possible the synchronous case might throw for more/different reasons than the asynchronous case because it "gets further along" the process of sending it, but I don't think we care.  The user code just needs to be written to catch exceptions on both the asynchronous fam_xxx() call and the fam_xxx_wait() call.

Reply:
Alright, that's sounds reasonable. We should be good to proceed here.